src/pudl/metadata/classes.py

```diff
@@ -1185,8 +1305,18 @@ def _check_harvest_primary_key(cls, value, info: ValidationInfo):
         return value
 
     @staticmethod
-    def dict_from_id(x: str) -> dict:  # noqa: C901
-        """Construct dictionary from PUDL identifier (`resource.name`).
+    def dict_from_id(resource_id: str) -> dict:
```

Comment:
It was surprisingly simple to run all of our `RESOURCE_METADATA` through the new type and validation machinery!

Reply:
Yes! Basically:
* It's helpful to have something like `PudlResourceDescriptor` to explicitly encode our bespoke resource descriptor data structure!
* It's... much less helpful if we don't actually enforce that that's what we're working with in the rest of the code.
* This should also blow up on incorrectly defined resources at validation time, instead of when we're then trying to do all this transformation logic to them.